src/lib/components/Swap/index.tsx

```diff
@@ -12,7 +14,38 @@ import Settings from './Settings'
 import SwapButton from './SwapButton'
 import Toolbar from './Toolbar'
 
-export default function Swap() {
+interface DefaultTokenAmount {
+  address?: string | { [chainId: number]: string }
+  amount?: number
+}
+
+interface SwapDefaults {
+  tokenList: string | TokenList['tokens']
```

Comment:
Are you sure `TokenList['tokens']` is what we want here?

Reply:
`TokenList['tokens'] === TokenInfo[]`, which is what we want. I didn't realize the equivalence until after writing this line before, so you see it as `TokenInfo[]` elsewhere. Will update.